### ISSUE TITLE
Accessibility - Locale Selector - Focus order issue in keyboard.

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/ibm-dotcom-library#1766]
## Detailed description

Issues :- 
1.Close icon
As soon as model dialog opens focus is not set on the first interactive element.

2.Shift-tab navigation
Doing Shift-tab navigation from close icon focus is going to the parent page

3.Select region(Location and languages page)
With tab/shift-tab focus is not going to the "Select region".

Expected result -
1.Once the overlay open, focus should set on the first interactive element of the page which is close icon
2.With Tab/shift tab navigation focus should stay within the model page. It should not go the parent page
3.Focus should go to each and every interactive element and it should be operable by keyboard

**Comment by RaphaelAmadeu** [comment]
@Praveen-sr 
Could you please test this again?
Thank you!

**Comment by Praveen-sr** [comment]
@RaphaelAmadeu 
1. Close icon :-
**On Storybook** - we don't have a triggering element here so issue cant be checked.
![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/81ab0547-99c1-4856-9f5b-40a458f367e8)
**On Learn Page** - when we open the locale selector from the footer region, model page will open and focus should set on the first interactive element which is close icon.

2. Shift-tab navigation
**On Storybook** - When we press Tab/shift tab multiple time  and observe focus is going to the parent page and not trapped within the model page. (Not working)
**On Learn Page** - when we open the locale selector from the footer region, pressing tab key focus comes to close icon and pressing tab key from there focus is going to parent page. (Not working)

3. Select region(Location and languages page)

**On Storybook** and **Learn Page** - When we hitting tab key, focus is set to the search field and doing shift tab key from search field, focus is going to the Select region back button.
Partially fixed:
(This issue is fixed now but we are getting a dotted line visual focus instead of thick and strong visual focus.)
![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/24b59ae0-6b47-4ccb-a898-4035d8421ed8)
